code/game/machinery/harvester.dm

```diff
@@ -75,21 +75,21 @@
 /obj/machinery/harvester/proc/can_harvest()
 	if(!powered() || state_open || !occupant || !iscarbon(occupant))
 		return
-	var/mob/living/carbon/C = occupant
+	var/mob/living/carbon/carbon_occupant = occupant
 	if(!allow_clothing)
-		for(var/A in C.held_items + C.get_equipped_items())
-			if(!isitem(A))
+		for(var/occupant_item in carbon_occupant.held_items + carbon_occupant.get_equipped_items())
+			if(!isitem(occupant_item))
 				continue
-			var/obj/item/I = A
-			if(!(HAS_TRAIT(I, TRAIT_NODROP)))
+			var/obj/item/abiotic_item = occupant_item
```

Comment:
we can do this better, just replace the whole typecheck with something that's written in the forloop

```dm
		for(var/obj/item/abiotic_item in carbon_occupant.held_items + carbon_occupant.get_equipped_items())
```

builtin typecheck is tres bon